packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -383,16 +381,10 @@ class _BottomNavigationTile extends StatelessWidget {
     // (which is an integer) by a large number.
     int size;
 
-    final BottomNavigationBarThemeData bottomTheme = BottomNavigationBarTheme.of(context);
-
     final double selectedFontSize = selectedLabelStyle.fontSize!;
 
-    final double selectedIconSize = selectedIconTheme?.size
-      ?? bottomTheme.selectedIconTheme?.size
-      ?? iconSize;
-    final double unselectedIconSize = unselectedIconTheme?.size
-      ?? bottomTheme.unselectedIconTheme?.size
-      ?? iconSize;
+    final double selectedIconSize = selectedIconTheme?.size ?? iconSize;
```

Comment:
Why are we now ignoring `bottomTheme.selectedIconTheme?.size`? Here and on the next line.

Reply:
We discussed this offline, but to summarize, the whole `BottomNavigationBarThemeData.selectedIconTheme` is ignored if `BottomNavigationBar.selectedIconTheme` is present. I updated the documentation on `BottomNavigationBarThemeData` to make that more clear.